common/flatpak-run.c

```diff
@@ -3092,8 +3092,94 @@ flatpak_run_add_app_info_args (GPtrArray      *argv_array,
   return TRUE;
 }
 
+static gboolean
+symlink_target_is_deployed (GFile *runtime_files, const char *symlink, const char *target)
+{
+  const char *runtime_files_path = NULL;
+  g_autofree char *deployed_target = NULL;
+
+  /* Manually check whether the symlink would point to a resolvable path inside the sandbox. */
+  runtime_files_path = flatpak_file_get_path_cached (runtime_files);
+  if (target[0] == '/')
+    {
+      /* Symlink points to an absolute path */
+      guint base_index = (g_strcmp0 (target, "/usr") == 0) ? 4 : 1;
+      deployed_target = g_strdup_printf ("%s/%s", runtime_files_path, &target[base_index]);
+    }
+  else
+    {
+      g_autofree char *symlink_subdir = NULL;
+      symlink_subdir = g_path_get_dirname (symlink);
+
+      if (!g_str_has_prefix (target, "../"))
```

Comment:
This is all pretty fragile. Symlink targets can be more complicated. For instance, it could easily be "./.." or suchlike.
I would recommend that you do something like:
base_file = g_file_new_from_path (symlink_subdir);
target_file = g_file_resolve_relative_path (base_file, target);
target_canonical = g_file_get_path (target_file);
if (g_str_has_prefix (target_canonical, "/usr"))
 => make it a symlink
else
 => bind-mount
symlink = g_file_new_from_path ()